src/schema/users.ts

```diff
@@ -1905,6 +1921,11 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       try {
         const updatedUser = { ...user, ...data, image: avatar };
         updatedUser.email = updatedUser.email?.toLowerCase();
+
+        if ('flags' in data) {
+          updatedUser.flags = { ...user.flags, ...data.flags };
+        }
```

Comment:
Difference with this and `updateFlagsStatement` is that if update to user happens between fetching the flags from the db and saving those flags would be lost:

1. repo.findOneBy(user)
2. some other logic updates user flags in the db
3. we merge the flags from 1. with input
4. repo.save
5. flags from 2. are overwritten

There is an exact pattern in `updateUserSettings` which can be reused, I pushed it in 1ff17ae3e7951fb940afbc3868fb17fa2d101589 for sanity check